src/Categories/Functor/Profunctor.agda

```diff
@@ -89,19 +91,46 @@ module Profunctor {o ℓ e} {o′} (C : Category o ℓ e) (D : Category o′ ℓ
     where
       open module F = Functor F
 
-  -- each Prof(C,D) is a category
-  homProf : (C : Category o ℓ e) → (D : Category o′ ℓ e) → Category _ _ e
-  homProf C D = record
-    { Obj = Profunctor C D
-    ; _⇒_ = λ P Q → NaturalTransformation P Q
-    ; _≈_ = λ α β → {!    !} -- equality of nat transf
-    ; id = Categories.NaturalTransformation.id
-    ; _∘_ = _∘ᵥ_
-    ; assoc = λ {A} {B} {C₂} {D₂} {f} {g} {h} → {!   !}
-    ; sym-assoc = {!   !}
-    ; identityˡ = {!   !}
-    ; identityʳ = {!   !}
-    ; identity² = {!   !}
-    ; equiv = {!   !}
-    ; ∘-resp-≈ = {!   !}
-    }
\ No newline at end of file
+-- each Prof(C,D) is a category
+homProf : {o o′ ℓ e : Level} → (C : Category o ℓ e) → (D : Category o′ ℓ e) → Category _ _ _
+homProf {o} {o′} {ℓ} {e} C D = record
+  { Obj = Profunctor C D
+  ; _⇒_ = λ P Q → NaturalTransformation P Q
+  ; _≈_ = _≃_
+  ; id = id
+  ; _∘_ = _∘ᵥ_
+  ; assoc     = λ { {f = f} {g} {h} → assoc-lemma {f = f} {g} {h}}
+  ; sym-assoc = λ { {f = f} {g} {h} → sym-assoc-lemma {f = f} {g} {h}}
+  ; identityˡ = λ { {f = f} → id-lemmaˡ {f = f}}
+  ; identityʳ = λ { {f = f} → id-lemmaʳ {f = f} }
+  ; identity² = λ z → z
+  ; equiv = ≃-isEquivalence
+  ; ∘-resp-≈ = λ { {f = f} {h} {g} {i} eq eq' → ∘ᵥ-resp-≃ {f = f} {h} {g} {i} eq eq' }
+  }
+  where
```

Comment:
Can't these 1-liners not be inlined in the above?  As `D` is visible, you could
```
  where
    module D = Category Y
```
and then everything is quite simple.

Reply:
We thought about inlining these lemmas, but we can't seem to get it to work without generalizing those vertical composition properties over any two categories. The implicit arguments on the 1-liners should be:

```agda
  ; assoc     = λ { {f = f} {g} {h} → assoc-lemma     {C = Product (Category.op D) C} {D = Setoids ℓ e} {f = f} {g} {h}}
  ; sym-assoc = λ { {f = f} {g} {h} → sym-assoc-lemma {C = Product (Category.op D) C} {D = Setoids ℓ e} {f = f} {g} {h}}
  ; identityˡ = λ { {f = f} → id-lemmaˡ {C = Product (Category.op D) C} {D = Setoids ℓ e} {f = f} }
  ; identityʳ = λ { {f = f} → id-lemmaʳ {C = Product (Category.op D) C} {D = Setoids ℓ e} {f = f} }
```

but unfortunately directly substituting in the lemmas does not work. If you manage to find some other way to remove them let us know!

Reply:
How did this end?

Reply:
It is 'ending' in me pulling it in as is (once the build passes). It wasn't clear to me that assoc (etc) were being called 'at' such different categories. So the current setup is just as nice as the one with the implicits spelled out.